src/hooks/usePlaceOrder.ts

```diff
@@ -22,11 +23,11 @@ interface Result {
 }
 
 export const usePlaceOrder = (): Result => {
-  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useSafeState(false)
   const { userAddress } = useWalletConnection()
 
-  const placeOrder = useMemo((): ((params: PlaceOrderParams) => Promise<boolean>) => {
-    return async ({ buyAmount, buyToken, sellAmount, sellToken }: PlaceOrderParams): Promise<boolean> => {
+  const placeOrder = useCallback(
+    () => async ({ buyAmount, buyToken, sellAmount, sellToken }: PlaceOrderParams): Promise<boolean> => {
```

Comment:
Almost
```suggestion
   async ({ buyAmount, buyToken, sellAmount, sellToken }: PlaceOrderParams): Promise<boolean> => {
```

and no `placeOrder()` on L79

Reply:
Aha! I thought `useCallback` needed to have 0 params. Updated.